src/mbgl/renderer/fill_bucket.cpp

```diff
@@ -35,7 +35,11 @@ FillBucket::~FillBucket() {
 }
 
 void FillBucket::addGeometry(const GeometryCollection& geometry) {
-    for (const auto& polygon : classifyRings(geometry)) {
+    // Max number of rings to allow per polygon.
+    // Optimizes polygons with many interior rings for earcut tesselation.
+    unsigned maxRings = 500;
```

Comment:
Let's use `uint32_t` type instead of `unsigned` everywhere — it's more predictable across platforms.